pkg/externalresource/external_resource.go

```diff
@@ -0,0 +1,65 @@
+package externalresource
+
+import (
+	b64 "encoding/base64"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/afero"
+)
+
+// ExternalResources represents the map of external resources at a manifest
+type ExternalResourceSection map[string]*ExternalResource
+
+// ExternalResource represents information on an external resource
+type ExternalResource struct {
+	Icon      string
+	Notes     Notes
+	Endpoints []ExternalEndpoint
+}
+
+// Notes represents information about the location and content of the external resource markdown
+type Notes struct {
+	Path     string
+	Markdown string // base64 encoded content of the path
+}
+
+// ExternalEndpoint represents information about an endpoint
+type ExternalEndpoint struct {
+	Name string
+	Url  string
+}
+
+// ERFilesystemManager represents ExternalResource information with the filesystem inyected
+type ERFilesystemManager struct {
+	ExternalResource ExternalResource
+	Fs               afero.Fs
+}
+
+// SetDefaults creates the necessary environment variables given an external resource
+func (er *ExternalResource) SetDefaults(externalName string) {
+	sanitizedExternalName := sanitizeForEnv(externalName)
+	for _, endpoint := range er.Endpoints {
+		sanitizedEndpointName := sanitizeForEnv(endpoint.Name)
+		endpointUrlEnv := fmt.Sprintf("OKTETO_EXTERNAL_%s_ENDPOINTS_%s_URL", sanitizedExternalName, sanitizedEndpointName)
+		os.Setenv(endpointUrlEnv, endpoint.Url)
+	}
+}
+
+func sanitizeForEnv(name string) string {
```

Comment:
what about spaces and special characters? does this name need to be part of any kubernetes resource?

Reply:
this sanitization is to incorporate the name of the external and its endpoints as an environment variable. But yes, it would be necessary to avoid blank spaces